flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointStatsTracker.java

```diff
@@ -632,4 +669,18 @@ public Long getValue() {
             }
         }
     }
+
+    private class LatestCompletedCheckpointTimeSinceMillisGauge implements Gauge<Long> {
+        @Override
+        public Long getValue() {
+            CompletedCheckpointStats completed = latestCompletedCheckpoint;
+            final long startTime;
+            if (completed != null) {
+                startTime = completed.getCompletedTimestamp();
```

Comment:
As discussed offline, ideally we might want some different logic if there are no resources available to run the job, e.g.:
```
Failed to trigger checkpoint for job xxx since Checkpoint triggering task yyy is not being executed at the moment. 
Aborting checkpoint. Failure reason: Not all required tasks are currently running..
```

I'm not sure how to implement this though.